src/appenders/components/FileAppender.ts

```diff
@@ -127,8 +127,14 @@ export class FileAppender extends BaseAppender {
    * @param options
    * @returns {streams.RollingFileStream}
    */
-  private openTheStream(file: string, fileSize: number | undefined, numFiles: number, options: any) {
-    const stream = new streams.RollingFileStream(file, fileSize, numFiles, options);
+  private openTheStream(file: string, fileSize: number | undefined, numFiles: number, pattern: string | undefined, options: any) {
+    let stream = null;
+    if (pattern) {
+      stream = new streams.DateRollingFileStream(file, pattern, options);
+    }
+    {
```

Comment:
Else is missing no ?

Reply:
My bad, I'll fix it and push into my repo.